lib/sierra-models/base.js

```diff
@@ -156,24 +156,74 @@ class SierraBase {
     return matches
   }
 
+  /**
+   *  Given an array of VarFieldMatch objects, returns a new array of matches
+   *  with dupes removed. A dupe entry is:
+   *   - An entry where there's another entry with an identical primary and
+   *     parallel value
+   *   - Orphaned parallel where there exists another entry with an identical
+   *     parallel that also has a primary value
+   *   - Orphaned primary where there exists another entry with an identical
+   *     primary that also has a parallel
+   */
   _uniqueVarFieldMatches (matches) {
     // Create a hash function, which will be used to determine equality of two
     // different maches. We consider two VarFieldMatches as being equal if
     // their primary and parallel values are equal:
     const hasher = (entry) => [entry.value, entry.parallel?.value].join('🎸')
+    let unique = uniqueObjectsByHash(matches, hasher)
```

Comment:
Can you add some documentation to the `uniqueObjectsByHash` function? 
My understanding is that it takes pairs of extracted/transformed values from an array of objects, and the object those are extracted from, and returns only objects for which the extracted/transformed values are unique. Is that accurate?
It also might be useful to describe what is being returned from `uniqueObjectsByHash` using the hasher function provided here.